plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/impl/SwitchTransitionDirection.java

```diff
@@ -0,0 +1,49 @@
+package org.yakindu.sct.refactoring.refactor.impl;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.gef.EditPart;
+import org.yakindu.sct.model.sgraph.Transition;
+import org.yakindu.sct.model.sgraph.Vertex;
+import org.yakindu.sct.refactoring.refactor.AbstractRefactoring;
+import org.yakindu.sct.ui.editor.editparts.TransitionEditPart;
+
+/* 
+ * TODO: This is not really a Refactoring. A Refactoring changes stuff without changing what it does, which does
+ * not apply on this "refactoring". There does not seem to be a better place for this at the moment, so when something like
+ * Modifications exists, this should be moved there.
+ */
```

Comment:
You are right this is not a refactoring but a model modification. In fact, we expose an API for model modifications that is used for example for the 'add outgoing state' action.
Take a look at org.yakindu.sct.ui.editor.modifications.ISemanticModification and AbstractSemanticModification.
There is also an adapter for both APIs RefactoringBasedModification that is used for Inline and Extract Subdiagram.